pkg/prometheus/prometheus.go

```diff
@@ -7,11 +7,9 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"google.golang.org/grpc"
+	"k8s.io/client-go/transport"
```

Comment:
It looks like this package is coming from k8s' client-go library but we are using this in the prometheus package. What is the reasoning behind switching this from the custom type `WrapTransport` to the k8s provided transport? I would think that we'd still want to keep using `WrapTransport` so we have control over the type.

Reply:
I agree that having this prometheus package depend on client-go is a bit weird.

Reply:
curiously, this change was motivated because our `prometheus.WrapTransport` type was no longer compatible with client-go's `config.WrapTransport` type:
```
pkg/k8s/api.go:48:23: cannot use prometheus.ClientWithTelemetry("k8s", wt) (type prometheus.WrapTransport) as type "k8s.io/client-go/transport".WrapperFunc in assignment
pkg/k8s/api.go:48:56: cannot use wt (type "k8s.io/client-go/transport".WrapperFunc) as type prometheus.WrapTransport in argument to prometheus.ClientWithTelemetry
```

config.go in client-go changed from `v10.0.0` to `v12.0.0`:
```diff
-WrapTransport func(rt http.RoundTripper) http.RoundTripper
+WrapTransport transport.WrapperFunc
```

...with this in `v12.0.0`:
```go
type WrapperFunc func(rt http.RoundTripper) http.RoundTripper
```

i agree the client-go dependency in our prometheus package is not great, i'll find a way to remove that.